File: ex232.java
import java.util.ArrayList;
import java.util.List;

class MyQueue {

    List<Integer> l = new ArrayList<>();
    List<Integer> t = new ArrayList<>();

    /** Initialize your data structure here. */
    public MyQueue() {

    }
    
    /** Push element x to the back of queue. */
    public void push(int x) {
        l.add(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int pop() {
        // Save other elements to a new list
        int l_size = l.size();
        for (int i = 0; i < l_size - 1; i++) {
            int temp = l.remove(l.size() - 1);
            t.add(temp);
        }

        // Remove the target element
        int r = l.remove(l.size() - 1);
        
        // Add those elements back to l
        int t_size = t.size();
        for (int i = 0; i < t_size; i++) {
            int temp = t.remove(t.size() - 1);
            l.add(temp);
        }

        return r;
    }
    
    /** Get the front element. */
    public int peek() {
        // Save other elements to a new list
        int l_size = l.size();
        for (int i = 0; i < l_size - 1; i++) {
            int temp = l.remove(l.size() - 1);
            t.add(temp);
        }

        // Get the target element
        int r = l.get(l.size() - 1);
        
        // Add those elements back to l
        int t_size = t.size();
        for (int i = 0; i < t_size; i++) {
            int temp = t.remove(t.size() - 1);
            l.add(temp);
        }

        return r;
    }
    
    /** Returns whether the queue is empty. */
    public boolean empty() {
        return l.size() == 0;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: ex232.cpp
class MyQueue {
public:
    stack<int> stk;
    int top = 0;    // Top element in the queue
    
    /** Initialize your data structure here. */
    MyQueue() {}
    
    /** Push element x to the back of queue. */
    void push(int x) {
        if (stk.empty())    top = x;
        stk.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        stack<int> stk2;
        while (stk.size() > 1) {
            stk2.push(stk.top());    stk.pop();
        }
        int r = stk.top();    stk.pop();
        
        if (!stk2.empty())    top = stk2.top();    // Reset the top value to second bottom element in the stack
        
        // Push elements back to stk
        while (!stk2.empty()) {
            stk.push(stk2.top());    stk2.pop();
        }
        
        return r;
    }
    
    /** Get the front element. */
    int peek() {
        return top;
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        return stk.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */